dip-template/runtimes/dip-consumer/src/dip.rs

```diff
@@ -30,20 +38,396 @@ use sp_std::{marker::PhantomData, vec::Vec};
 
 use crate::{weights, AccountId, DidIdentifier, Runtime, RuntimeCall, RuntimeOrigin};
 
-pub type MerkleProofVerifierOutput = <ProofVerifier as IdentityProofVerifier<Runtime>>::VerificationResult;
-/// The verifier logic assumes the provider is a sibling KILT parachain, the
-/// relaychain is a Rococo relaychain, and that a KILT subject can provide DIP
-/// proof that reveal at most 10 DID keys and 10 linked accounts (defaults
-/// provided by the `KiltVersionedParachainVerifier` type). Calls that do not
-/// pass the [`DipCallFilter`] will be discarded early on in the verification
-/// process.
-pub type ProofVerifier = KiltVersionedParachainVerifier<
+// +1 for the web3name.
+const MAX_PROVIDER_REVEALABLE_KEYS_COUNT: u32 = MAX_PUBLIC_KEYS_PER_DID + MAX_REVEALABLE_LINKED_ACCOUNTS + 1;
+
+/// The verifier logic is tied to the provider template runtime definition.
+pub type ProviderTemplateProofVerifier = KiltVersionedParachainVerifier<
 	RelaychainRuntime,
 	RelayStateRootsViaRelayStorePallet<Runtime>,
 	2_000,
 	ProviderRuntime,
 	DipCallFilter<KeyIdOf<ProviderRuntime>, BlockNumberFor<ProviderRuntime>, ProviderAccountId>,
+	(),
+	DEFAULT_MAX_PROVIDER_HEAD_PROOF_LEAVE_COUNT,
+	DEFAULT_MAX_PROVIDER_HEAD_PROOF_LEAVE_SIZE,
+	DEFAULT_MAX_DIP_COMMITMENT_PROOF_LEAVE_COUNT,
+	DEFAULT_MAX_DIP_COMMITMENT_PROOF_LEAVE_SIZE,
+	DEFAULT_MAX_DID_MERKLE_PROOF_LEAVE_COUNT,
+	DEFAULT_MAX_DID_MERKLE_PROOF_LEAVE_SIZE,
+	MAX_PROVIDER_REVEALABLE_KEYS_COUNT,
 >;
+pub type MerkleProofVerifierInput = <ProviderTemplateProofVerifier as IdentityProofVerifier<Runtime>>::Proof;
+pub type MerkleProofVerifierOutput =
+	<ProviderTemplateProofVerifier as IdentityProofVerifier<Runtime>>::VerificationResult;
+// Wrapper around the verifier to implement the `GetWorstCase` trait.
+pub struct ProviderTemplateProofVerifierWrapper;
+
+// Delegate verification logic to the specialized version of
+// `KiltVersionedParachainVerifier`.
+impl IdentityProofVerifier<Runtime> for ProviderTemplateProofVerifierWrapper {
+	type Error = <ProviderTemplateProofVerifier as IdentityProofVerifier<Runtime>>::Error;
+	type Proof = MerkleProofVerifierInput;
+	type VerificationResult = MerkleProofVerifierOutput;
+
+	fn verify_proof_for_call_against_details(
+		call: &pallet_dip_consumer::RuntimeCallOf<Runtime>,
+		subject: &<Runtime as pallet_dip_consumer::Config>::Identifier,
+		submitter: &<Runtime as frame_system::Config>::AccountId,
+		identity_details: &mut Option<<Runtime as pallet_dip_consumer::Config>::LocalIdentityInfo>,
+		proof: Self::Proof,
+	) -> Result<Self::VerificationResult, Self::Error> {
+		<ProviderTemplateProofVerifier as IdentityProofVerifier<Runtime>>::verify_proof_for_call_against_details(
+			call,
```

Comment:
Reviewers can skip this altogether, as it's been generated in a test setup and copy-pasted here. The unit test below is more meaningful to review.